File: Python3/프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def isRight(s) -> bool:
    slist = list()
    for c in s:
        if c == '(' or c == '[' or c == '{': #열린 괄호일 때
            slist.append(c)
        elif bool(slist) and (slist[-1] == '(' and c == ')' or slist[-1] == '{' and c == '}' or slist[-1] == '[' and c == ']'):
            slist.pop()
        else: return False
    return len(slist) == 0
    
    
def solution(s):
    answer = 0
    for c in s:
        s = s[1:] + s[0]
        answer += int(isRight(s))
    return answer

File: C++/프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.cpp
#include <string>
#include <vector>
#include <stack>
#include <iostream>

using namespace std;

bool findPair(string s) {
    stack<char> stack;
    for(char c : s) {
        if(c == '(' || c == '{' || c == '[') {
            stack.push(c);
        } else if(!stack.empty() && (stack.top() == '(' && c == ')' || stack.top() == '{' && c == '}' || stack.top() == '[' && c == ']')) {
            stack.pop();
        } else {
            return false;
        }
    }
    return stack.empty();
}

int solution(string s) {
    int answer = 0;
    for(auto& c : s) {
        s += s[0];
        s.erase(s.begin());
        answer += findPair(s);
    }
    return answer;
}